src/Parsers/ParserDropQuery.cpp

```diff
@@ -54,11 +55,15 @@ bool parseDropQuery(IParser::Pos & pos, ASTPtr & node, Expected & expected, cons
     else if (s_all.ignore(pos, expected) && s_tables.ignore(pos, expected) && kind == ASTDropQuery::Kind::Truncate)
     {
         has_all_tables = true;
+        if (s_from.ignore(pos, expected))
+        {
+            if (s_if_exists.ignore(pos, expected))
+                if_exists = true;
 
-        if (s_if_exists.ignore(pos, expected))
-            if_exists = true;
-
-        if (!name_p.parse(pos, database, expected))
+            if (!name_p.parse(pos, database, expected))
+                return false;
+        }
+        else
             return false;
```

Comment:
I think this will be enough:
```
        if (s_from.ignore(pos, expected))
            return false;

        if (s_if_exists.ignore(pos, expected))
            if_exists = true;

        if (!name_p.parse(pos, database, expected))
            return false;
```